### PR TITLE
Support max_item_size for query frontend memcached cache

### LINKED ISSUE #3895: Supporting max item size for the memcached client (query-frontend)
### Is your proposal related to a problem?

I'm seeing error messages from the query-frontend relating to memcached insertions failing:

```
{
  "caller": "memcached.go:235",
  "err": "server=10.252.1.8:11211: memcache: unexpected response line from \"set\": \"SERVER_ERROR object too large for cache\r\n\"",
  "level": "error",
  "msg": "failed to put to memcached",
  "name": "",
  "ts": "2021-03-03T06:16:55.594447923Z"
}
```

We configure max item sizes of 16MB for both the query-range and labels caches, and set the corresponding max item size on the memcached servers.

This appears to be due to a known incomplete feature, as I also see log messages "MaxItemSize is not yet supported by the memcached client" that appear to originate from https://github.com/thanos-io/thanos/pull/3114. The code has a TODO above it to add this feature: https://github.com/thanos-io/thanos/blob/513707d368b3910db8cf4744cf7df48abf176a04/pkg/queryfrontend/config.go#L101-L104

### Describe the solution you'd like

For the memcached clients used by query-frontend to support max item size.

### Describe alternatives you've considered

n/a

### Additional context

I don't see similar error messages from the store, for which we've deployed index and chunk memcached caches. I've only glanced at the code related to this, but the store's client doesn't appear to have this problem. We're using the default 1MB max item size there.

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixes #3895

This pr upgrades Cortex and support `max_item_size` in query frontend Memcached config.

### CODE REVIEW

File: docs/components/query-frontend.md [hunk 1]
```diff
@@ -85,7 +85,11 @@ config:
   expiration: 0s
 ```
 
-`expiration` specifies memcached cache valid time , If set to 0s, so using a default of 24 hours expiration time
+`expiration` specifies memcached cache valid time.  If set to 0s, so using a default of 24 hours expiration time.
+
+If a `set` operation is skipped because of the item size is larger than `max_item_size`, this event is tracked by a
+
```
Comment (pracucci) [inline docs/components/query-frontend.md:91]: I think this newline is unintentional.

**Review by pracucci — APPROVED** [review]
LGTM (modulo a nit). Good job! 👏

**Comment by kakkoyun** [comment]
Hey @yeya24. We're going to cut a release candidate soon, is there any chance for you to rebase and push this to the finish line?

**Comment by yeya24** [comment]
@kakkoyun Thank you. Please take a look.

**Review by kakkoyun — APPROVED** [review]
LGTM 🚀

**Pull request merged**